program.sh

```diff
@@ -1,17 +1,47 @@
 #!/bin/sh
 
+# 10 min interval
+INTERVAL=600
+STOP=stop
+
 download_repository() {
-	echo "swift download $1"
-	swift download $1
+    echo "swift download $1"
+    swift download $1
 }
 
 upload_repository() {
-	swift upload $1 repository/ sync/
+    swift upload $1 repository/ sync/
 } 
 
-trap 'kill -TERM $PID; wait $PID; upload_repository $1; exit 143' TERM
+watch_upload() {
+    trap 'kill -TERM $PIDSLEEP' SIGTERM
+
+    while ! [ -f $STOP ]
+    do
+	sleep $INTERVAL &
+	PIDSLEEP=$!
+	wait $PIDSLEEP
+	date
+	upload_repository $1
+    done
+}
+
+before_exit() {
```

Comment:
I also need some comments to explain this function.

Reply:
The purpose of this function is to tell to the main process the rest of the tasks to do before exiting.  
And to exit, the process has to in order:
- terminate the `learn-ocaml` instance process
- terminate the watcher process

Du to the fact that I can not terminate the watcher process suddently, I tell the watcher to loop while a file `stop` does not exist, and I create the file in the `before_exit function`